datasync/chaindatafetcher/kafka/consumer.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package kafka
+
+import (
+	"context"
+
+	"github.com/Shopify/sarama"
+)
+
+type Consumer struct {
+	cancel   chan bool
+	handler  map[string]func(*sarama.ConsumerMessage) error
+	consumer sarama.ConsumerGroup
+	ctx      context.Context
+	isActive bool
+}
+
+func NewConsumer(ctx context.Context, consumer sarama.ConsumerGroup) *Consumer {
+	return &Consumer{
+		cancel:   make(chan bool),
+		handler:  map[string]func(*sarama.ConsumerMessage) error{},
+		ctx:      ctx,
+		consumer: consumer,
+		isActive: false,
+	}
+}
+
+func (r *Consumer) Subscribe(topic string, handler func(*sarama.ConsumerMessage) error) error {
```

Comment:
`r` for `Consumer` type looks confusing.

Reply:
Renamed the receiver to `c`. 1f989f2dcfbb1c9eeed58c88629d79559f246f26